src/lesson/lesson.service.ts

```diff
@@ -1,26 +1,113 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 import { CreateLessonDto } from './dto/create-lesson.dto';
 import { UpdateLessonDto } from './dto/update-lesson.dto';
+import { LessonInterface } from './interfaces/lesson.interface';
+import { UserInterface } from '../user/interfaces/user.interface';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { User } from '../user/entities/user.entity';
 
 @Injectable()
 export class LessonService {
-  create(createLessonDto: CreateLessonDto) {
-    return 'This action adds a new lesson';
+  constructor(
+    @InjectModel('Lesson') private readonly lessonModel: Model<LessonInterface>,
+    @InjectModel('User') private readonly userModel: Model<UserInterface>, // @InjectModel('Group') private readonly groupModel: Model<GroupInterface>,
+  ) {}
+
+  async create(createLessonDto: CreateLessonDto) {
+    const lesson = await new this.lessonModel(createLessonDto);
+    return lesson.save();
+  }
+
+  async findAll(): Promise<LessonInterface[]> {
+    const lessons = await this.lessonModel
+      .find()
+      .populate({ path: 'students', model: User, select: '-password' })
+      .populate({ path: 'teachers', model: User, select: '-password' })
+      .exec();
+    if (!lessons || !lessons[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return lessons;
   }
 
-  findAll() {
-    return `This action returns all lesson`;
+  async findByID(id: string): Promise<LessonInterface> {
+    const lesson = await this.lessonModel
+      .findOne({ _id: id })
+      .populate({ path: 'students', model: User, select: '-password' })
+      .populate({ path: 'teachers', model: User, select: '-password' })
+      .exec();
+    if (!lesson) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return lesson;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} lesson`;
+  async findAllByDate(date: Date): Promise<LessonInterface[]> {
+    const lessons = await this.lessonModel.find({ date: date }).exec();
+    if (!lessons || !lessons[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return lessons;
   }
 
-  update(id: number, updateLessonDto: UpdateLessonDto) {
-    return `This action updates a #${id} lesson`;
+  async findAllByTeacherID(userID: string): Promise<LessonInterface[]> {
+    const lessons = await this.lessonModel
+      .find({
+        teachers: {
+          $elemMatch: { $in: { _id: userID } },
```

Comment:
wooow, ale się ciekawych rzeczy dowiedziałam ;D ale na moje oko i z tego co wyczytałam w dokumentacji, to `$in` jest tutaj zbędne, bo jest używane do tablic i wtedy zwraca dany dokument jeśli nawet jedna wartość z tej tablicy jest prawdą, a tutaj jest tylko pojedyncze `_id`, więc samo `$elemMatch` by wystarczyło, a nawet i bez tego powinno działać bo jest to jako [Single Query Condition](https://docs.mongodb.com/manual/reference/operator/query/elemMatch/#elemmatch-single-query-condition). Czy mam zły tok myślenia? 

I dzięki! Bo nie wiedziałam że Mongo są takie rożne operatory

Reply:
zostawiam

Reply:
oki, jednak dobrze jest